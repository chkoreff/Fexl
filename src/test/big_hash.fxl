#
evaluate
(
use_lib "format.fxl";
use_lib "time.fxl";
use "test.fxl";
standard
) \;

# Reference: https://www.di-mgt.com.au/sha_testvectors.html
# Test the extremely-long message.

# About 12.0s to buffer.
\text=(show_microtime "buffer";
repeat_str
"abcdefghbcdefghicdefghijdefghijkefghijklfghijklmghijklmnhijklmno" 16777216)

# Test sha512.
(
# About 4.4s to hash.
\hash=(show_microtime "sha512"; sha512 text)
say "sha512"
put "text = " say "(the extremely-long message)"
put "hash = " show_hex hash
nl
)
# Test sha256.
(
# About 7.2s to hash.
\hash=(show_microtime "sha256"; sha256 text)
say "sha256"
put "text = " say "(the extremely-long message)"
put "hash = " show_hex hash
nl
)
