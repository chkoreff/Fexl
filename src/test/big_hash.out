time buffer = 9961378 us
time sha512 = 4049748 us
sha512
text = (the extremely-long message)
hash = 64:[b47c933421ea2db149ad6e10fce6c7f93d0752380180ffd7f4629a712134831d77be6091b819ed352c2967a2e2d4fa5050723c9630691f1a05a7281dbe6c1086]

time sha256 = 6294375 us
sha256
text = (the extremely-long message)
hash = 32:[50e72a0e26442fe2552dc3938ac58658228c0cbfb1d2ca872ae435266fcd055e]

